make r_extralight rescale the 0-255 range into r_extralight-255 instead of simply adding

RicardoLuis0 opened the pull request:
this lets the range of values it can work with be greatly extended without harming lighting effects on maps

RicardoLuis0 committed:
make r_extralight rescale the 0-255 range into r_extralight-255 instead of simply adding

RicardoLuis0 commented:
(also changes the allowed range from `-8, +8` to `-64, +128`)

Boondorl commented on src/rendering/swrenderer/scene/r_light.cpp:
```diff
@@ -147,7 +148,7 @@ namespace swrenderer
 	fixed_t LightVisibility::LightLevelToShadeImpl(RenderViewport *viewport, int lightlevel, bool foggy)
 	{
-		lightlevel += r_extralight;
+		lightlevel = hw_ClampLight(lightlevel);
```
Is this safe? The original sw renderer doesn't clamp to 255, at least not this early in, that seemed to be a hw specific thing. I'm not sure what knock on effects this could have

RicardoLuis0 replied:
removed the clamp from software mode

RicardoLuis0 committed:
rename hw_ClampLight to RescaleLightLevel, remove clamp from software

RicardoLuis0 committed:
fix wrong function

visual studio refactoring missed this one, i should've double checked what it was doing before commiting 🤦‍♀️
